### PR TITLE
Lobby system for new joining players

### PR DESCRIPTION
**Pull request by sp00ktober**
This adds a lobby to the game. When starting a new game players can join into the galaxy selection screen now. Joining players will land in that screen too if:

- Host is still in galaxy selection
- Player joins the first time on that savegame

The join normally into the game if they already played as clients on that savegame.

What i plan to add:

- players can choose their starting planet in the galaxy selection screen.
- players can see planet details when clicking on a planet in galaxy selection screen

This will be especially usefull regarding the combat system that the game devs want to add since this could be the base for a player vs player scenario.

Here is a short demo how it looks atm.

[Joining the lobby on a new game](https://youtu.be/ekLnA7mnpf4)
[Joining the lobby on a running game](https://youtu.be/L8qZYAKoam4)
[Joining the game after you already did as client](https://youtu.be/F2CNlzesOJU)

### CODE REVIEW

**Comment by sp00ktober** [comment]
grafik
Solar system details can now be visualized, but there are still some things i need to implement here.

Also this is currently not compatible with GS2 as it prefixes methods in UIVirtualStarmap blocking the patches i added.

**Comment by sp00ktober** [comment]
Compatibility with GS2 is done through https://github.com/Touhma/DSP_Plugins/pull/146

**Comment by sp00ktober** [comment]
grafik
Last commit added the possibility to view planet details (vein amount not accurate with GS2 yet) as well as moving "back" in detail level by clicking into space.

**Comment by sp00ktober** [comment]
Should be okay now, there are some slight QoL things that could be added but i leave that for later to focus more on the reported bugs.

File: NebulaNetwork/Client.cs [hunk 1]
```diff
@@ -246,7 +252,10 @@ private void ClientSocket_OnClose(object sender, CloseEventArgs e)
         private static void DisableNagleAlgorithm(WebSocket socket)
         {
             TcpClient tcpClient = AccessTools.FieldRefAccess<WebSocket, TcpClient>("_tcpClient")(socket);
-            tcpClient.NoDelay = true;
+            if (tcpClient != null)
+            {
+                tcpClient.NoDelay = true;
+            }
```
Comment (PhantomGamers) [inline NebulaNetwork/Client.cs:258]: was this done for a particular reason? Is tcpClient null when opening the connection for the lobby? i think it shouldn't be hm

Comment (sp00ktober) [reply NebulaNetwork/Client.cs:258]: If it was for a particular reason i dont remember it anymore :)

**Review by PhantomGamers — APPROVED** [review]
Looks good to me, works great! Great work! :D

**Review by hubastard — APPROVED** [review]

**Comment by hubastard** [comment]
Nice work !

**Pull request merged**